### PR TITLE
[docs]Created a doc for preparing an app for TestFlight

### PR DESCRIPTION
**Pull request by dustin-wojciechowski**
I created a short doc to help anyone uploading their app to Test Flight. 

I targeted MacCatalyst apps for the Mac App Store but would work the same for iOS apps.

### CODE REVIEW

File: docs/preparing-your-app-for-testflight.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+There is some preparation involved with preparing your Apple Developer account for publishing, such as preparing an App Identifier and a Provisioning Profile. The MAUI documentation has a great guide for that:
+https://aka.ms/maui-publish-app-store
+
+Also, your project will need to have configuration values set in the .csproj file, the Info.plist file, and the Entitlements.plist file.
+
+## Preparing your Project File (.csproj) for App Store submission
+
+Before you can submit your app to the App Store, you need to make sure that it is correctly configured. This includes:
+
```
Comment (stephen-hawley) [inline docs/preparing-your-app-for-testflight.md:13]: Each of these should have tell where and how to set these and or an explanation that answers the question "how do I know it's correct?"

Maybe insert an annotated sample .csproj with each of these items set to a correct value?

File: docs/preparing-your-app-for-testflight.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+The `Info.plist` file is used by the build system to configure your app and is already generated by your project. It contains info such as the Bundle Display Name and Bundle Identifier.
+### Entitlements.plist
+
+The `Entitlements.plist` file is used by the build system to sign your app.
+
+Some of the values in the `Entitlements.plist` file are set by the build system. These values are:
+
+- com.apple.security.app-sandbox
+- com.apple.security.network.client
+
```
Comment (stephen-hawley) [inline docs/preparing-your-app-for-testflight.md:33]: What settings might you need to make and why?

File: docs/preparing-your-app-for-testflight.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+## Building your App for Publishing
+
+Once you've set all the correct values, you can build your app for publishing. You can build your app for publishing very easily by using the dotnet build command. Here is an example for building a MacCatalyst app with .NET 7.0:
```
Comment (stephen-hawley) [inline docs/preparing-your-app-for-testflight.md:36]: "you can build your app for publishing. You can build your app for publishing" - redundant.
Maybe "build your app for publishing. You can do this by using..."

File: docs/preparing-your-app-for-testflight.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+`dotnet build -f:net7.0-maccatalyst -c:Release`
+
+## App Store Connect
+
+App Store Connect is the web site that you use to manage your apps on the App Store. You can use App Store Connect to upload your app to the App Store, and you can use App Store Connect to manage your app once it is on the App Store.
```
Comment (stephen-hawley) [inline docs/preparing-your-app-for-testflight.md:42]: suggest: "App Store Connect is the App Store management site. You can use it to upload apps to the App and and to manage your apps."

**Review by stephen-hawley — COMMENTED** [review]
Dustin - I approached this by strictly looking at the language and ensuring that it's crisp and clear. Editing in this way is a subjective process, but I think this will help the overall flow.

File: docs/preparing-your-app-for-testflight.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+3. Click on the "+" button.
+4. Enter the name of your app.
+5. Select the platform for your app.
+6. Select the primary language for your app.
+7. Select the category for your app. Note: It is important that the primary category you select here will be the one 
+8. Click on "Create".
+
+### Uploading your App using Transporter
+
+Once you have created an app on App Store Connect, you can upload your app to the App Store. You can upload your app to the App Store by following these steps:
```
Comment (tj-devel709) [inline docs/preparing-your-app-for-testflight.md:59]:
```suggestion
Once you have created an app on App Store Connect, you can upload your app to the App Store by following these steps:
```

Comment (tj-devel709) [reply docs/preparing-your-app-for-testflight.md:59]: Just a suggestion for conciseness - can be applied to the below examples if wanted

**Review by tj-devel709 — APPROVED** [review]
This is awesome! I learned a lot on the process from the review! I made a small suggestion and I think after Steve's suggestions are reviewed, this PR LGTM!

**Review by mandel-macaque — APPROVED** [review]
No native speaker, but the process is correct.

File: docs/preparing-your-app-for-testflight.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
+# Preparing your App for TestFlight
+
+Publishing to the App Store requires some preparation. This guide should help you get started with using Transporter and the App Connect site to upload your app to the App Store for internal testing via TestFlight. This guide will use the example of publishing a MacCatalyst app to the Mac App Store, but the process is similar for iOS and tvOS apps.
+
+There is some preparation involved with preparing your Apple Developer account for publishing, such as preparing an App Identifier and a Provisioning Profile. The MAUI documentation has a great guide for that:
+https://aka.ms/maui-publish-app-store
+
+Also, your project will need to have configuration values set in the .csproj file, the Info.plist file, and the Entitlements.plist file.
```
Comment (rolfbjarne) [inline docs/preparing-your-app-for-testflight.md:8]: This sounds better to me:

```suggestion
Also, your project will need to have certain configuration values set in the .csproj file, the Info.plist file, and the Entitlements.plist file.
```

File: docs/preparing-your-app-for-testflight.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+- Setting the correct `ApplicationId` in your project.
+- Setting the correct `ApplicationVersion` in your project.
+- Setting the correct `CodesignKey` in your project.
+- Setting the correct `PackagingSigningKey` in your project.
+- Setting the correct `CodesignEntitlements` in your project.
+- Setting the correct `CodesignProvision` in your project.
+
+### Info.plist
+
+The `Info.plist` file is used by the build system to configure your app and is already generated by your project. It contains info such as the Bundle Display Name and Bundle Identifier.
```
Comment (rolfbjarne) [inline docs/preparing-your-app-for-testflight.md:24]: The bundle identifier can either be set with the CFBundleIdentifier key in the Info.plist file, or the ApplicationId in the project file. This sentence seems to indicate that you have to set CFBundleIdentifier in the Info.plist, which isn't correct if ApplicationId is set in the project file (as mentioned in he previous section).

There are a few other configuration values that can either be set in the project file or the Info.plist file - more info about that here: https://github.com/xamarin/xamarin-android/blob/main/Documentation/guides/OneDotNetSingleProject.md

File: docs/preparing-your-app-for-testflight.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+### Entitlements.plist
+
+The `Entitlements.plist` file is used by the build system to sign your app.
+
+Some of the values in the `Entitlements.plist` file are set by the build system. These values are:
+
+- com.apple.security.app-sandbox
```
Comment (rolfbjarne) [inline docs/preparing-your-app-for-testflight.md:31]: Which values depend on the platform and what the app does.

For example: com.apple.security.app-sandbox only applies to macOS and Mac Catalyst.

Might want to link to Apple's documentation: https://developer.apple.com/documentation/bundleresources/entitlements?language=objc

**Review by rolfbjarne — APPROVED** [review]

**Comment by dustin-wojciechowski** [comment]
Not sure I'll be able to address any more comments until after my leave, but if looks good feel free to merge :)

**Pull request merged**